_posts/2024-09-12-v5-release.md

```diff
@@ -0,0 +1,139 @@
+the bleeding edge (even if it is a rather dull edge) then please consider waiting until we move to `latest` to upgrade. That said, we look forward to working with you all as you
+upgrade to support and address any bugs you encounter.
+
+## The Breaking Changes
+
+As I said above, we did the minimum number of breaking changes we could. I try here to list them in order of impact to application owners, but there are a fair number of subtle
+chnages burried in here which you should read the changelog for more details on.
+
+### Goodbye Node.js 0.10, Hello Node 18 and up.
+
+We dropped support for Node.js versions lower than 18. This was a long time coming, but it is probably the most important change for us as maintainers of these libraries. Keeping old
+Node.js version support was holding us back from many critical performance and maintainability changes. Now that we have dropped these versions we have more stable and maintainable
+CI, we can start adopting some newer language and runtime features, and we can drop many dependencies which are no longer required.
+
+We recognize that this might mean some enterprises have difficulty with older or "parked" applications, and because of this we are working on a partnership with HeroDevs to offer
+"Never Ending Support" which will include critical security patches even after v4 enters End-of-Life (more on these plans will come soon). That said, we strongly suggest that folks
+update to modern Node.js versions as soon as possible.
+
+### Path Matching and Regular Expressions
+
+We updated from `path-to-regexp@0.x` to `path-to-regexp@8.x` in v5, needless to say there were many years of changes in this. If you were using any of the 5.0.0-beta releases, we
+landed a last minute update which greatly changed the path semantics so we could [remove any ReDoS chances going forward](https://blakeembrey.com/posts/2024-09-web-redos/). For
+more detailed changes, [see the `path-to-regexp` readme](https://github.com/pillarjs/path-to-regexp?tab=readme-ov-file#express--4x).
+
+#### 1. No more regex. We dropped support for sub-expression regular expressions (ex. `/:foo(\\d+)`).
+
+This is one of the most common patterns in wide use which we have removed. The unfortunate nature of regular expressions is how easy it is to write one with exponential time behaviors
+when parsing input. The dreaded ReDoS. It turns out it is very difficult to prevent users from doing this, and as a library which converts strings to regular expressions, we are on
+the hook for the security aspects of this.
+
+*How to migrate:* We recommend using more robust input validation libraries. [There are many on `npm`](https://www.npmjs.com/search?q=validate%20express) depending on your needs.
+Shameless plug from the author, I maintain [a middleware based "code first" OpenAPI library](https://www.npmjs.com/package/@wesleytodd/openapi) for this kind of thing. 
```

Comment:
I do not mind first person switchs, it was written by a person lol. That said, I want to give users a clear recommendation, not a wishy wasy "there are options", but since I have never used the other options I cannot vouch for them to make a recommendation. Happy to take a poll of the team and see which ones folks have had success with and recommend a list of them, but I would rather not tell an audience with mixed technical backgrounds "use a validation lib" with no guidance on what that means.